test/main.test.js

```diff
@@ -129,4 +129,49 @@ describe('LoopBack Context', function() {
       ], done);
     });
   });
+  it('doesn\'t mix up contexts if using concurrently then() from when 3.7.7',
+  function() {
+    var timeout = 50;
+    // Concurrent execution number 1 of 2
+    var execution1 = new Promise(function execution1(outerResolve, reject) {
+      LoopBackContext.runInContext(function pushToContext1() {
+        var ctx = LoopBackContext.getCurrentContext();
+        expect(ctx).is.an('object');
+        ctx.set('test-key', 'test-value-1');
+        var whenPromise = whenV377.promise(function(resolve) {
+          setTimeout(resolve, timeout);
```

Comment:
Can we simplify this code by using `promise.delay`? See https://github.com/cujojs/when/blob/master/docs/api.md#promisedelay